llvm/lib/Target/RISCV/MCA/RISCVCustomBehaviour.cpp

```diff
@@ -185,6 +187,59 @@ RISCVInstrumentManager::createInstruments(const MCInst &Inst) {
   return SmallVector<UniqueInstrument>();
 }
 
+/// Return EMUL = (EEW / SEW) * LMUL
+inline static std::pair<unsigned, bool>
+getEMULEqualsEEWDivSEWTimesLMUL(unsigned EEW, unsigned SEW,
+                                RISCVII::VLMUL VLMUL) {
+  // Calculate (EEW/SEW)*LMUL preserving fractions less than 1. Use GCD
+  // to put fraction in simplest form.
+  auto [LMUL, Fractional] = RISCVVType::decodeVLMUL(VLMUL);
```

Comment:
Can this function be simpler? For example, use `getSEWLMULRatio` (please verify the functionality):
```cpp
int Ratio = RISCVVType::getSEWLMULRatio(SEW, VLMUL);
int LMulFixedPoint = (EEW * 8) / Ratio;
bool Fractional = LMulFixedPoint < 8;
unsigned LMul = Fractional ? 8 / LMulFixedPoint : LMulFixedPoint / 8;
```

Reply:
This doesn’t seem to work, but I will play with it and see if I can figure out how to do it using fixed point approach.

Reply:
I have updated this function to no longer use GCD approach. However, I did not figure out how to do this calculation using fixed point. Please feel free to suggest an alternative solution to the one I am now proposing.

Reply:
Why is fixed point approach not working? I tried it, it can calculate the right EMUL, see #69570.

Reply:
I'm not sure what happened. I gave it a try and saw changes when I reran update_mca_test_checks. However, I gave it another try and it works fine. I will approve your PR since it comes with tests and is in a file where it can be reused. I will rebase this patch once merged. Thanks for the suggestion!!